### PR TITLE
[Clang] avoid adding consteval condition as the last statement to preserve valid CFG

### LINKED ISSUE #116485: clang doesn't warn for missing return value if function contains "if consteval"
Compile this little snippet, and notice that clang doesn't give a warning for missing `return` (if the `if consteval` is removed, then the warning is properly shown).

```c++
int foo() {
    if consteval {
    }
}
```

version: `Debian clang version 20.0.0 (++20241008102056+4d218caa7716-1~exp1`

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #116485

### CODE REVIEW

File: clang/test/SemaCXX/constexpr-return-non-void-cxx2b.cpp [hunk 1]
```diff
@@ -5,3 +5,9 @@ static_assert(__is_same(decltype([] constexpr -> int { }( )), int)); // expected
 
 consteval int g() { } // expected-warning {{non-void function does not return a value}}
 static_assert(__is_same(decltype([] consteval -> int { }( )), int)); // expected-warning {{non-void lambda does not return a value}}
+
+namespace GH116485 {
+int h() {
+    if consteval { }
+} // expected-warning {{non-void function does not return a value}}
```
Comment (AaronBallman) [inline clang/test/SemaCXX/constexpr-return-non-void-cxx2b.cpp:12]: Other test cases to add are:
```
void foo(int x) {
    if consteval {
    }
    switch (x) {
    case 1:
      foo(12); // Should get a `-Wimplicit-fallthrough` warning here
    case 2:
      break;
    }
}
```
https://godbolt.org/z/rns3Yjofq

or testing whether the consteval if actually causes an early return:
```
constexpr bool foo()  {
    if !consteval { return true; }
}

bool b = foo();
constinit bool c = foo();
```

**Review by AaronBallman — COMMENTED** [review]
The changes generally LGTM, but I did have additional tests to try.

File: clang/test/SemaCXX/constexpr-return-non-void-cxx2b.cpp (hunk 1 above)
Comment (a-tarasyuk) [reply clang/test/SemaCXX/constexpr-return-non-void-cxx2b.cpp:12]: @AaronBallman thanks for the feedback. I’ve added tests, and the first one works as expected. Could you check the last one to see if I need to add any extra handling for that case?

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Pull request merged**